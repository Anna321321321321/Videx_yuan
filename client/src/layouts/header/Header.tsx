import { Col, Layout, Row } from 'antd';
import React from 'react';
import { Link, browserHistory } from 'react-router';
import * as Logger from 'videx/client/logger';
import UserContainer from '../../container/user-container';

export default () => (
  <Layout.Header className="videx-header">
    <Row>
      <Col lg={4} md={6} sm={8} xs={10}>
        <Link
          className="videx-header-logo"
          onClick={() => Logger.event('App.Home.Click')}
          to="/"
        >
          <img
            className="videx-header-logo-img"
            alt="logo"
            src="/static/logo.png"
          />
          <span className="videx-header-logo-span">ViDeX</span>
        </Link>
      </Col>
      <Col lg={20} md={18} sm={16} xs={14}>
        <div className="videx-header-avatar">
          <UserContainer />
        </div>
      </Col>
    </Row>
  </Layout.Header>
);
