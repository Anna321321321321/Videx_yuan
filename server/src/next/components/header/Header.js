import { Layout } from 'antd';
import React, { Component } from 'react';

export default class Header extends Component {
  render() {
    return (
      <Layout.Header
        style={{
          transition: 'opacity 0.5s',
          background: '#fff',
          marginBottom: '24px',
          padding: '0 48px',
          width: '100%',
          display: 'flex',
          flexDirection: 'row',
          alignItems: 'center'
        }}
      >
        <img
          alt="logo"
          src="/static/logo.png"
          style={{
            height: '31px',
            lineHeight: '31px',
            marginRight: '8px'
          }}
        />
        <span
          style={{
            fontSize: '16px',
            fontFamily: "'Raleway', 'Hiragino Sans GB', sans-serif"
          }}
        >
          ViDeX
        </span>
      </Layout.Header>
    );
  }
}
