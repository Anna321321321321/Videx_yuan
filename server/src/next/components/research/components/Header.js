import { Layout, Menu, Button, Icon } from 'antd';
import React, { Component, Fragment } from 'react';
import MediaQuery from 'react-responsive';

export default class Header extends Component {
  state = {
    collapsed: false
  };
  toggleCollapsed = () => {
    this.setState({
      collapsed: !this.state.collapsed
    });
  };

  render() {
    return (
      <Layout.Header
        style={{
          transition: 'opacity 0.5s',
          background: '#fff',
          marginBottom: '24px',
          padding: '0 48px',
          width: '100%',
          display: 'flex',
          flexDirection: 'row',
          alignItems: 'center',
          position: 'fixed',
          zIndex: 99
        }}
        breakpoint={'lg'}
      >
        <img
          alt="logo"
          src="/static/logo.png"
          style={{
            height: '31px',
            lineHeight: '31px',
            marginRight: '8px'
          }}
        />
        <MediaQuery minWidth={350}>
          <span
            style={{
              fontSize: '16px',
              fontFamily: "'Raleway', 'Hiragino Sans GB', sans-serif",
              textTransform: 'uppercase'
            }}
          >
            ViDeX Research
          </span>
        </MediaQuery>

        <MediaQuery minWidth={767}>
          <Menu
            theme="light"
            mode="horizontal"
            style={{ lineHeight: '40px', marginLeft: 'auto' }}
            onClick={item => (window.location.href = item.key)}
          >
            <Menu.Item key="#home">
              <Icon type="home" />Home
            </Menu.Item>
            <Menu.Item key="#researchArea">
              {' '}
              <Icon type="pie-chart" />Research areas
            </Menu.Item>
            <Menu.Item key="#publications">
              {' '}
              <Icon type="file-pdf" />Publications
            </Menu.Item>
            <Menu.Item key="#people">
              {' '}
              <Icon type="team" />People
            </Menu.Item>
            <Menu.Item key="#collaborate">
              {' '}
              <Icon type="phone" />Collaborate
            </Menu.Item>
          </Menu>
        </MediaQuery>
        <MediaQuery maxWidth={767}>
          <div style={{ position: 'relative', marginLeft: 'auto' }}>
            <Button
              onClick={this.toggleCollapsed}
              style={{ marginBottom: 16, marginLeft: 'auto' }}
            >
              <Icon type={this.state.collapsed ? 'menu-unfold' : 'menu-fold'} />
            </Button>
            {this.state.collapsed && (
              <Menu
                theme="light"
                mode="inline"
                onClick={item => (window.location.href = item.key)}
                inlineCollapsed={!this.state.collapsed}
                style={{
                  width: 170,
                  position: 'absolute',
                  top: '50px',
                  right: '-30px'
                }}
              >
                <Menu.Item key="#home">
                  {' '}
                  <Icon type="home" />
                  <span>Home</span>
                </Menu.Item>
                <Menu.Item key="#researchArea">
                  {' '}
                  <Icon type="pie-chart" />
                  <span>Research areas</span>
                </Menu.Item>
                <Menu.Item key="#publications">
                  {' '}
                  <Icon type="file-pdf" />
                  <span>Publications</span>
                </Menu.Item>
                <Menu.Item key="#people">
                  {' '}
                  <Icon type="team" />
                  <span>People</span>
                </Menu.Item>
                <Menu.Item key="#collaborate">
                  <Icon type="phone" />
                  <span>Collaborate</span>
                </Menu.Item>
              </Menu>
            )}
          </div>
        </MediaQuery>
      </Layout.Header>
    );
  }
}
