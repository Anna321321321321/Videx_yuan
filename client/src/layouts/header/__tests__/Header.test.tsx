import { shallow } from 'enzyme';
import React from 'react';
import Header from '..';

describe('Header Layout Unit Tests', () => {
  it('Render component', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });
});
