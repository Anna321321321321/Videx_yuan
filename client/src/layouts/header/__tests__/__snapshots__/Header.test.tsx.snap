// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Layout Unit Tests Render component 1`] = `
<Adapter
  className="videx-header"
>
  <Row
    gutter={0}
  >
    <Col
      lg={4}
      md={6}
      sm={8}
      xs={10}
    >
      <Link
        className="videx-header-logo"
        onClick={[Function]}
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/"
      >
        <img
          alt="logo"
          className="videx-header-logo-img"
          src="/static/logo.png"
        />
        <span
          className="videx-header-logo-span"
        >
          ViDeX
        </span>
      </Link>
    </Col>
    <Col
      lg={20}
      md={18}
      sm={16}
      xs={14}
    >
      <div
        className="videx-header-avatar"
      >
        <Connect(UserContainer) />
      </div>
    </Col>
  </Row>
</Adapter>
`;
